app/[locale]/community/notice/[id]/page.tsx

```diff
@@ -1,12 +1,36 @@
+import { Metadata } from 'next';
 import { Suspense } from 'react';
 
+import { getNoticePostDetail } from '@/apis/notice';
+
 import PostFallback from '@/components/layout/fallback/PostFallback';
 import PageLayout from '@/components/layout/pageLayout/PageLayout';
 
 import { PostSearchQueryParams } from '@/types/post';
 
+import { getMetadata } from '@/utils/metadata';
+import { notice } from '@/utils/segmentNode';
+
 import NoticeViewer from './NoticeViewer';
 
+export async function generateMetadata({
+  params: { locale, id },
+  searchParams,
+}: {
+  params: { locale: string; id: string };
+  searchParams: PostSearchQueryParams;
+}): Promise<Metadata> {
```

Comment:
이 파일 처럼 `NoticePostPageProps`등으로 기존 Prop 타입이 이미 있는 경우에는 그걸 가져와서 쓰는거 어때? locale은 기존 타입에 추가하면 될 거 같고 지금은 프롭 정보가 중복되어있어서 수정 사항이 있으면 두군데를 수정해야될거같아서

```suggestion
export async function generateMetadata({
  params: { locale, id },
  searchParams,
}: NoticePostPageProps) {
```